runtime/lua/vim/treesitter/dev.lua

```diff
@@ -76,6 +76,9 @@ local function traverse(node, depth, lang, injections, tree)
     })
 
     traverse(child, depth + 1, lang, injections, tree)
+
+    -- Add closing parens to final child
+    tree[#tree].text = string.format('%s)', tree[#tree].text)
```

Comment:
If you want to simplify this for the anonymous nodes, you could put the end parenthesis on a new line. If you don't like that kind of formatting, I think it makes sense to put it like this when there is no anonymous nodes at the end, and put it after the last anonymous node otherwise.